flake.nix

```diff
@@ -2,64 +2,51 @@
+
     systems.url = "github:nix-systems/default";
+    flake-parts.url = "github:hercules-ci/flake-parts";
   };
 
-  outputs = { self, flake-utils, nixpkgs-unstable, nixpkgs, ... }:
-    flake-utils.lib.eachDefaultSystem
-      (system:
-        let
-          pkgs = nixpkgs.legacyPackages.${system};
-          pkgs-unstable = nixpkgs-unstable.legacyPackages.${system};
-          versions = builtins.fromJSON (builtins.readFile ./versions.json);
-        in
-        {
-          devShells.default = pkgs.mkShell {
-            buildInputs = [
-              pkgs-unstable.black
-              (pkgs-unstable.python3.withPackages
-                (ps: [
-                  ps.pygithub
-                  ps.semver
-                ])
-              )
-              pkgs-unstable.nix-prefetch
-              pkgs.nodejs
-              pkgs.rubyPackages.dotenv
-            ];
-          };
-          # https://github.com/NixOS/nix/issues/7165
-          checks = self.packages.${system};
-          packages = builtins.mapAttrs
-            (version: { hash, vendorHash }: self.lib.buildTerraform {
-              inherit system version hash vendorHash;
-            })
-            versions;
-        }) // {
-      lib.buildTerraform = { system, version, hash, vendorHash }:
-        let
-          pkgs = nixpkgs.legacyPackages.${system};
-          pkgs-unstable = nixpkgs-unstable.legacyPackages.${system};
-        in
-        # https://www.hashicorp.com/blog/hashicorp-adopts-business-source-license
-        if builtins.compareVersions version "1.6.0" >= 0
-        then
-        # https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/applications/networking/cluster/terraform/default.nix
-          pkgs-unstable.mkTerraform
-            {
-              inherit version hash vendorHash;
-              patches = [ "${nixpkgs-unstable}/pkgs/applications/networking/cluster/terraform/provider-path-0_15.patch" ];
-            }
-        else
-        # https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/applications/networking/cluster/terraform/default.nix
-          pkgs.mkTerraform {
-            inherit version hash vendorHash;
-            patches = [ "${nixpkgs}/pkgs/applications/networking/cluster/terraform/provider-path-0_15.patch" ];
-          };
+  outputs = { self, flake-parts, ... }@inputs: flake-parts.lib.mkFlake { inherit inputs; } {
+    imports = [
+      inputs.flake-parts.flakeModules.easyOverlay
+    ];
+    systems = import inputs.systems;
+
+    perSystem = { config, pkgs, pkgs-unstable, system, ... }: {
+      _module.args = {
+        pkgs = import inputs.nixpkgs {
+          inherit system;
+        };
+        pkgs-unstable = import inputs.nixpkgs-unstable {
+          inherit system;
+          config.allowUnfree = true;
+        };
+      };
+
+      packages = import ./packages.nix { inherit pkgs pkgs-unstable; custom-lib = self.lib; };
+
+      overlayAttrs = {
+        terraform-versions = config.packages;
```

Comment:
I don't like the name, but couldn't think of anything better.
This allows users to access the packages like: `pkgs.terraform-versions."1.6.0"`.

Quick repl demo:
```nix
nix-repl> :lf .
Added 23 variables.

nix-repl> pkgs = import inputs.nixpkgs { system = "x86_64-linux"; overlays = [ outputs.overlays.default ]; }

nix-repl> builtins.attrNames pkgs.terraform-versions
[ "1.0.0" "1.0.1" "1.0.10" "1.0.11" "1.0.2" "1.0.3" "1.0.4" "1.0.5" "1.0.6" "1.0.7" "1.0.8" "1.0.9" "1.1.0" "1.1.1" "1.1.2" "1.1.3" "1.1.4" "1.1.5" "1.1.6" "1.1.7" "1.1.8" "1.1.9" "1.2.0" "1.2.1" "1.2.2" "1.2.3" "1.2.4" "1.2.5" "1.2.6" "1.2.7" "1.2.8" "1.2.9" "1.3.0" "1.3.1" "1.3.10" "1.3.2" "1.3.3" "1.3.4" "1.3.5" "1.3.6" "1.3.7" "1.3.8" "1.3.9" "1.4.0" "1.4.1" "1.4.2" "1.4.3" "1.4.4" "1.4.5" "1.4.6" "1.4.7" "1.5.0" "1.5.1" "1.5.2" "1.5.3" "1.5.4" "1.5.5" "1.5.6" "1.5.7" "1.6.0" "1.6.1" "1.6.2" "1.6.3" "1.6.4" "1.6.5" "1.6.6" "1.7.0" "1.7.1" "1.7.2" ]


```

Reply:
@pedorich-n I can't think of a better name either; `terraform-versions` seems reasonable for the time being.